### DISCUSSION TITLE
'Throughput energy [W.h]' in solution summary_variables

### DISCUSSION THREAD

**Discussion by ldy5413** [pybamm-team/PyBaMM#4114] [Q&A] [answered]
Hi, I am new to pybamm and trying to do some aging simulation. I would like to use the 'Throughput energy [W.h]' in solution summary_variables but it always shows 0 even though I set 'calculate discharge energy' to be 'true' in model options. It seems enable this feature only calculate throughput energy in solution.cycles. Am I missing some model options or is there a way to directly access throughput energy in summary_variables as I don't want to save detailed cycle data to save ram. Below is a demo code to reproduce my problem.
```
import pybamm
options = {
    'calculate discharge energy': "true"
}
model = pybamm.lithium_ion.DFN(options=options)
params = pybamm.ParameterValues("OKane2022")

experiments = pybamm.Experiment([(
            f"Discharge at 1 C until 2.75 V",
            "Rest for 0.1 hour",
            f"Charge at 0.5 C until 4.2 V",
            f"Hold at 4.2 V until 0.05 C",
            "Rest for 0.1 hour",
        )] * 10)
sim = pybamm.Simulation(model, parameter_values=params, experiment=experiments)
solution = sim.solve()
solution.summary_variables['Throughput energy [W.h]']
```

**Comment by Thiyaga2025** [comment]
hi, 
"solution['Throughput energy [W.h]'].entries" gives the array and "solution['Throughput energy [W.h]'].entries[-1]" gives the final value. sorry if i have not understood your question correctly.

**Reply by ldy5413** [reply]
hi,
thanks for the answer, I tried your method and I found solution['Throughput energy [W.h]'].entries did give out some throughput energy array. I have one more question regarding to this, I found I have to save all cycles to make it store all the data, is there a way to use solution.summary_variables? The reason that I use solution.summary_variables instead of solution is summary_variables always give array with the length of cycles so that I can easliy configure an intepolator for capacity and energy throughput for aging optimization and I can save ram to run more case in parallel.

**Reply by Thiyaga2025** [reply, marked as answer]
hi @ldy5413 , were you able to get 'Throughput energy [W.h]' in summary variables. i am able to get the values at the end of pybamm simulation.

**Reply by ldy5413** [reply]
yes, I am able to get the values of 'Throughput energy [W.h]' in summary variables after I update the pybamm to latest version.
